File: Car.ts
class Car {

	//Attributes
	public brandArray: string[] = [ "Opel", "Ferrari", "Porsche", "BMW" ];
	private brand: string;
	private consumption: number;
	private tankvolume: number;
	liter: number;
	km: number;

	
	//Constructor(s)
	public constructor(brand: string, consumption: number, tankvolume: number ){
        this.setBrand(brand);
		this.setConsumption(consumption);
		this.setTankvolume(tankvolume);
	}
	//Methods
	
	
	//setter
	public setBrand (newBrand: string): void {
		this.brand = newBrand;
	}
	public vsetLiter(newLiter: number): void {
		this.liter = newLiter;
	}
	public setKm(newKm: number): void {
		this.km = newKm;
	}
	public setConsumption(newConsumption: number): void {
		this.consumption = newConsumption;
	}
	public setTankvolume(newTankvolume: number): void {
		this.tankvolume = newTankvolume;
	}
		
	
	//getter
		public getBrand(): string {
		return this.brand;
	}
	public getLiter(): number {
		return this.liter;
	}
	public getKm(): number {
		return this.km;
	}
	public getConsumption(): number {
		return this.consumption;
	}
	public getTankvolume(): number {
		return this.tankvolume;
	}
	public getAllBrands(): string {
		for (let s of this.brandArray) {           
			//Do your stuff here
			console.log(s + " ");
		}
		return this.brandArray.toString();
	}
	
		
	//others
	public calculateConsumption(liter: number, km: number): number {
		return this.getLiter()/this.getKm();
	}
	

}

File: CarreraGame/src/car/Car.java
package car;

import java.util.Arrays;

public class Car {

	//Attributes
	String[] brandArray = { "Opel", "Ferrari", "Porsche", "BMW" };
	String brand;
	private double consumption;
	private double tankvolume;
	double liter;
	double km;

	
	//Constructor(s)
	public Car(String brand, double consumption, double tankvolume ){
		this.setConsumption(consumption);
		this.setTankvolume(tankvolume);
	}
	//Methods
	
	
	//setter
	public void setBrand(String newBrand) {
		this.brand = newBrand;
	}
	public void setLiter(double newLiter) {
		this.liter = newLiter;
	}
	public void setKm(double newKm) {
		this.km = newKm;
	}
	public void setConsumption(double newConsumption) {
		this.consumption = newConsumption;
	}
	public void setTankvolume(double newTankvolume) {
		this.tankvolume = newTankvolume;
	}
		
	
	//getter
		public String getBrand() {
		return this.brand;
	}
	public double getLiter() {
		return this.liter;
	}
	public double getKm() {
		return this.km;
	}
	public double getConsumption() {
		return this.consumption;
	}
	public double getTankvolume() {
		return this.tankvolume;
	}
	public String getAllBrands() {
		for (String s: this.brandArray) {           
			//Do your stuff here
			System.out.print(s + " ");
		}
		return this.brandArray.toString();
	}
	
		
	//others
	public double calculateConsumption(double liter, double km) {
		return this.getLiter()/this.getKm();
	}
	


}
